bit/network/services.py

```diff
@@ -11,6 +14,104 @@ def set_service_timeout(seconds):
     DEFAULT_TIMEOUT = seconds
 
 
+class RPCHost:
+    def __init__(self, user, password, host, port, use_https):
+        self._session = requests.Session()
+        self._url = "http{s}://{user}:{password}@{host}:{port}/".format(
+            s="s" if use_https else "",
+            user=user,
+            password=password,
+            host=host,
+            port=port,
+        )
+        self._headers = {"content-type": "application/json"}
+        self._session.verify = use_https
+
+    def __getattr__(self, rpc_method):
+        return RPCMethod(rpc_method, self)
+
+    def get_balance(self, address):
+        getcontext().prec = 9
+        b = Decimal(self.getreceivedbyaddress(address, 0))
+        return int(float(str(b * 100000000)))
+
+    def get_balance_testnet(self, address):
+        return self.get_balance(address)
+
+    def get_transactions(self, address):
+        r = self.listreceivedbyaddress(0, True, True, address)
+        if len(r) > 0:
+            r = r[0]["txids"]
+        return r
+
+    def get_transactions_testnet(self, address):
+        return self.get_transactions(address)
+
+    def get_transaction_by_id(self, txid):
+        return self.getrawtransaction(txid, False)
+
+    def get_transaction_by_id_testnet(self, txid):
+        return self.get_transaction_by_id(txid)
+
+    def get_unspent(self, address):
+        r = self.listunspent(0, 9999999, [address])
+        return [
+            Unspent(
+                currency_to_satoshi(tx["amount"], "btc"),
+                tx["confirmations"],
+                tx["scriptPubKey"],
+                tx["txid"],
+                tx["vout"],
+            )
+            for tx in r
+        ]
+
+    def get_unspent_testnet(self, address):
+        return self.get_unspent(address)
+
+    def broadcast_tx(self, tx_hex):
+        try:
+            _ = self.sendrawtransaction(tx_hex)
+        except BitcoinNodeException:
```

Comment:
Would it be worth at least logging the exeption message as a warning? I'm not certain but curious what your thoughts are.

Reply:
Makes sense. I forgot about it but it is important to find out why a transaction eventually fails to be broadcast. Will add a logging warning!